source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -271,8 +272,8 @@ bool VaryHeader::isAllowed(const Http::ResponseHeaderMap& headers) const {
 }
 
 bool VaryHeader::hasVary(const Http::ResponseHeaderMap& headers) {
-  const Http::HeaderEntry* vary_header = headers.get(Http::Headers::get().Vary);
-  return vary_header != nullptr && !vary_header->value().empty();
+  const auto vary_header = headers.get(Http::Headers::get().Vary);
+  return !vary_header.empty() && !vary_header[0]->value().empty();
```

Comment:
so vary can definitely be multi-value.  While it's not reasonable to send
Vary:
Var: foo, bar
I think it's legal.

I wonder if we can just remove this utility and do a quick vary_header.empty() check in isAllowed for the performance wins?

Reply:
Will put a TODO in here for caching people to look at.

Reply:
Now that I look at this code more, I think a lot of this code would be vastly simpler if Vary where just a custom inline header, but that is getting into a larger change than I want to make, so I will just put that into the TODO.